### PR TITLE
test for wp_privacy_exports_url

### PR DESCRIPTION
**Pull request by pbearne**
Trac ticket: https://core.trac.wordpress.org/ticket/59709

### CODE REVIEW

**Comment by jdstphen00** [comment]
how can i do for this site
https://sloto-mania.co.uk
https://cryptocasino4u.com
https://365bet4u.com

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Tests for the wp_privacy_exports_url function.
+ *
+ * @group functions.php
+ *
+ * @covers ::wp_privacy_exports_url
+ */#
+class Tests_functions_test_wp_privacy_exports_url extends WP_UnitTestCase {
```
Comment (mindctrl) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: With the current filename and classname, phpunit complains about not being able to find the class. We can do the class or the file.

```suggestion
class Tests_functions_test_wpPrivacyExportsUrl extends WP_UnitTestCase {
```

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Tests for the wp_privacy_exports_url function.
+ *
+ * @group functions.php
+ *
+ * @covers ::wp_privacy_exports_url
+ */#
```
Comment (mindctrl) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]:
```suggestion
 */
```

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+class Tests_functions_test_wp_privacy_exports_url extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 59709
+	 */
+	public function test_wp_privacy_exports_url() {
+
+		$this->assertEquals( 'http://example.org/wp-content/uploads/wp-personal-data-exports/', wp_privacy_exports_url() );
```
Comment (mindctrl) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: Just a nit, to match other patterns in the tests folder.

```suggestion
		$this->assertSame( 'http://example.org/wp-content/uploads/wp-personal-data-exports/', wp_privacy_exports_url() );
```

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	/**
+	 * @ticket 59709
+	 */
+	public function test_wp_privacy_exports_url_filtered() {
+
+		add_filter( 'wp_privacy_exports_url', array( $this, 'filter_wp_privacy_exports_url' ) );
+
+		$expected_url = 'https://filtered.com/wp-personal-data-exports/';
+		$actual_url   = wp_privacy_exports_url();
+		$this->assertEquals( $expected_url, $actual_url );
```
Comment (mindctrl) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]:
```suggestion
		$this->assertSame( $expected_url, $actual_url );
```

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+		remove_filter( 'wp_privacy_exports_url', array( $this, 'filter_wp_privacy_exports_url' ) );
```
Comment (mindctrl) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: Maybe an edge case, but I think we need to move `remove_filter` to `tear_down()`, in case the assertion ever fails it won't leak into other tests.

**Review by mindctrl — CHANGES_REQUESTED** [review]

**Review by mindctrl — APPROVED** [review]

**Comment by masteradhoc** [comment]
Thanks for the work @pbearne and the reviews @mindctrl. i'll milestone this for 7.1 now.

**Review by mindctrl — COMMENTED** [review]
Looks good overall. The remaining Copilot suggestions make sense.

**Comment by t-hamano** [comment]
Hi @pbearne, do you have the bandwidth to address the Copilot reviews?

**Comment by pbearne** [comment]
done

**Comment by masteradhoc** [comment]
~~@pbearne can you recheck? the last commit removed the } on line 22.~~

LGTM! @t-hamano can you recheck this one as well?

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Tests for the wp_privacy_exports_url function.
+ *
+ * @group functions.php
```
Comment (t-hamano) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]:
```suggestion
 * @group functions
 ```

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @covers ::wp_privacy_exports_url
+ */
+class Tests_Functions_wpPrivacyExportsUrl extends WP_UnitTestCase {
+
+	public function tear_down() {
```
Comment (t-hamano) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: This `tear_down` method itself should be removable because anything added via `add_filter` should be cleaned up by the base class's tear_down.

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+	/**
+	 * Filter for test.
+	 *
+	 * @param string $url Exports directory URL.
+	 * @return string Filtered exports directory URL.
+	 */
+	public function filter_wp_privacy_exports_url( $url ) {
+
+		return 'https://filtered.com/wp-personal-data-exports/';
```
Comment (t-hamano) [inline tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]:
```suggestion
		return str_replace( 'wp-personal-data-exports/', 'filtered-exports/', $exports_url );
```

This is a similar change to the one in https://github.com/WordPress/wordpress-develop/pull/5553#discussion_r3698613696. I believe we need to test the filter parameters.

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php (hunk 6 above)
Comment (t-hamano) [reply tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: Fixed in 8a127b00fa

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php (hunk 7 above)
Comment (t-hamano) [reply tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: Fixed in d501f656b6

File: tests/phpunit/tests/functions/wpPrivacyExportsUrl.php (hunk 8 above)
Comment (t-hamano) [reply tests/phpunit/tests/functions/wpPrivacyExportsUrl.php]: Fixed in 4428823cfe

**Review by t-hamano — APPROVED** [review]
